packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js

```diff
@@ -35,45 +82,85 @@ class TemplateValidator extends BaseValidator {
 
   // Validates:
   // - Theme does not contain more than TEMPLATE_LIMIT templates
-  // - All templates have a "label" annotation
+  // - All templates have valid template types
+  // - All templates that require a label have a "label" annotation
+  // - All templates that require a screenshot have a "screenshotPath" annotation
   validate(absoluteThemePath, files) {
     let validationErrors = [];
-    const templates = files.filter(isTemplate);
-    const numTemplates = templates.length;
+    let templateCount = 0;
+
+    files.forEach(filePath => {
+      if (isCodedFile(filePath)) {
+        const annotations = getFileAnnotations(filePath);
+        const templateType = getAnnotationValue(
+          annotations,
+          ANNOTATION_KEYS.templateType
+        );
+        if (templateType) {
+          const label = getAnnotationValue(annotations, ANNOTATION_KEYS.label);
+          const screenshotPath = getAnnotationValue(
+            annotations,
+            ANNOTATION_KEYS.screenshotPath
+          );
+
+          // Ignore global partials and templates with type of none in count
+          if (!TEMPLATE_COUNT_IGNORE_LIST.includes(templateType)) {
+            templateCount++;
+          }
+
+          const validations = VALIDATIONS_BY_TYPE[templateType];
+
+          if (validations) {
+            if (!validations.allowed) {
+              validationErrors.push(
+                this.getError(this.errors.INVALID_TEMPLATE_TYPE, {
+                  templatePath: path.relative(absoluteThemePath, filePath),
+                  templateType,
+                })
+              );
+            }
+            if (validations.label && !label) {
+              validationErrors.push(
+                this.getError(this.errors.MISSING_LABEL, {
+                  templatePath: path.relative(absoluteThemePath, filePath),
+                })
+              );
+            }
+            if (validations.screenshot && !screenshotPath) {
+              validationErrors.push(
+                this.getError(this.errors.MISSING_SCREENSHOT_PATH, {
+                  templatePath: path.relative(absoluteThemePath, filePath),
+                })
+              );
+            }
+          } else {
+            logger.debug(
+              `Unrecognized template type "${templateType}" in ${path.relative(
+                absoluteThemePath,
+                filePath
+              )}`
+            );
+          }
+        } else {
+          logger.debug(
+            `No template type found for ${path.relative(
+              absoluteThemePath,
+              filePath
+            )}`
+          );
```

Comment:
Should these also be validation errors?

Reply:
I'm not sure, but maybe @KatzMitch will know. In this scenario the html file will have to have a `.html` extension, `isAvailableforNewContent !== false`, and no templateType annotation. 

somewhat related, I'm currently doing nothing for html files that have no annotations at all and I'm not sure if I should be showing some sort of validation message for those.

Reply:
Hmm since it's just specific to the marketplace, I think we should require that all HTML files have annotations and throw an error if they don't (cc: @ckconant )

Reply:
Yeah, I agree with @KatzMitch that all html files should have annotations. The only file that has limited annotations is the `base.html` layout in [boilerplate](https://github.com/HubSpot/cms-theme-boilerplate/blob/main/src/templates/layouts/base.html#L2), which only has `templateType: none` annotated.

Reply:
OK, I'll add an additional error type for this 👍

Reply:
added in a8db57d